networkit/cpp/centrality/GroupClosenessLocalSwaps.cpp

```diff
@@ -0,0 +1,362 @@
+// networkit-format
+
+#include <algorithm>
+#include <cassert>
+#include <omp.h>
+#include <queue>
+
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/centrality/GroupClosenessLocalSwaps.hpp>
+
+namespace NetworKit {
+
+GroupClosenessLocalSwaps::GroupClosenessLocalSwaps(const Graph &G, const std::vector<node> &group,
+                                                   count maxSwaps)
+    : GroupClosenessLocalSwaps(G, group.begin(), group.end(), maxSwaps) {}
+
+void GroupClosenessLocalSwaps::init() {
+    const auto n = G->upperNodeIdBound();
+
+    distance.assign(n, 0);
+    gamma.assign(n * group.size(), false);
+    canSwap.assign(group.size(), false);
+    visited.assign(n, false);
+
+    idxMap.clear();
+    idxMap.reserve(group.size());
+
+    stack.assign(n, 0);
+    farness.assign(group.size(), 0);
+    farnessDecrement.assign(group.size(), 0);
+    sumOfMins.assign(n, 0);
+
+    intDistributions.resize(omp_get_max_threads());
+    urngs.reserve(omp_get_max_threads());
+    for (index i = 0; i < static_cast<index>(omp_get_max_threads()); ++i)
+        urngs.emplace_back(Aux::Random::getURNG());
```

Comment:
This looks broken since the loop is not parallel and we get several identical references to the URNG of thread 0.

Reply:
I also wonder if this is necessary at all. `urngs` is only used in a `parallelForNodes` loop in `initRandomVector`. Accessing the thread-local variable is cheap compared to the loop. Why don't we just do the following in `initRandomVector`:
```cpp
#pragma omp parallel
{
    auto &urng = Aux::Random::getURNG();

    #pragma omp for
    for(node u = 0; u < G.upperNodeIdBound(); ++u) {
        [...]
    }
}